File: frontend/src/types/hospitalIntelligence.ts
export type DataSourceType = 'real' | 'estimated' | 'simulated' | 'unavailable'

export interface FieldValue {
  value: string | number | boolean | null
  data_source_type: DataSourceType
  source?: string
}

export interface HospitalClickPayload {
  name: string
  latitude: number
  longitude: number
  address?: string
  place_id?: string
  mapbox_id?: string
  patients_assigned?: number
  critical_assigned?: number
  non_critical_assigned?: number
  eta_minutes?: number
  distance_km?: number
  congestion?: string
  impact_zone?: string
}

export interface DoctorRecord {
  name: string
  specialty: string
  organization: string
  practice_location?: string
  affiliation_match?: 'at_hospital' | 'same_city' | 'other_area'
  npi: string
  data_source_type: DataSourceType
  source: string
}

export interface NurseCount {
  count: number
  data_source_type: DataSourceType
  source?: string
}

export interface EquipmentRecord {
  name: string
  count: number
  data_source_type: DataSourceType
  source: string
}

export interface HospitalIntelligenceProfile {
  hospital_profile: {
    hospital_id: string
    name: string
    address: string
    coordinates: { lat: number; lng: number }
    phone: string
    website: string
    hospital_type?: string
    emergency_department?: FieldValue
    data_confidence: 'high' | 'medium' | 'low'
    sources: string[]
  }
  capacity: {
    scenario_assignment: {
      critical_patients: FieldValue
      non_critical_patients: FieldValue
      total_patients: FieldValue
      eta_minutes: FieldValue
      distance_km: FieldValue
      congestion: FieldValue
      impact_zone: FieldValue
    }
    total_beds: FieldValue
    icu_beds: FieldValue
    emergency_services: FieldValue
    trauma_level: FieldValue
    occupancy: FieldValue
  }
  doctors: DoctorRecord[]
  nurses: {
    er_nurses: NurseCount
    icu_nurses: NurseCount
    trauma_nurses: NurseCount
    general_ward_nurses: NurseCount
  }
  equipment: EquipmentRecord[]
  ai_recommendation: {
    can_accept: {
      critical_patients: number
      moderate_patients: number
      minor_patients: number
    }
    resource_gaps: string[]
    routing_advice: string
    staffing_recommendations: string[]
    equipment_recommendations: string[]
  }
  collected_at?: string
  agent_status?: Record<string, string>
}

export function sourceBadge(type: DataSourceType) {
  switch (type) {
    case 'real':
      return { label: 'REAL', className: 'bg-emerald-500/20 text-emerald-200 border-emerald-400/30' }
    case 'estimated':
      return { label: 'ESTIMATED', className: 'bg-amber-500/20 text-amber-200 border-amber-400/30' }
    case 'simulated':
      return { label: 'SIMULATED', className: 'bg-violet-500/20 text-violet-200 border-violet-400/30' }
    default:
      return { label: 'UNAVAILABLE', className: 'bg-slate-500/20 text-slate-300 border-slate-400/30' }
  }
}

export async function fetchHospitalIntelligence(
  apiBase: string,
  payload: HospitalClickPayload
): Promise<HospitalIntelligenceProfile> {
  const response = await fetch(`${apiBase}/api/v1/hospital-intelligence`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(payload)
  })

  if (!response.ok) {
    const errorBody = await response.text()
    throw new Error(`Hospital intelligence failed (${response.status}): ${errorBody}`)
  }

  return response.json()
}

File: backend/app/schemas/hospital_intelligence.py
from typing import Any, Literal

from pydantic import BaseModel, Field


DataSourceType = Literal["real", "estimated", "simulated", "unavailable"]


class HospitalIntelligenceRequest(BaseModel):
    name: str
    latitude: float
    longitude: float
    address: str | None = None
    place_id: str | None = None
    mapbox_id: str | None = None
    patients_assigned: int | None = None
    critical_assigned: int | None = None
    non_critical_assigned: int | None = None
    eta_minutes: int | None = None
    distance_km: float | None = None
    congestion: str | None = None
    impact_zone: str | None = None


class FieldValue(BaseModel):
    value: Any | None = None
    data_source_type: DataSourceType = "unavailable"
    source: str = ""


class Coordinates(BaseModel):
    lat: float
    lng: float


class HospitalProfile(BaseModel):
    hospital_id: str
    name: str
    address: str = ""
    coordinates: Coordinates
    phone: str = ""
    website: str = ""
    hospital_type: str = ""
    emergency_department: FieldValue = Field(default_factory=FieldValue)
    data_confidence: Literal["high", "medium", "low"] = "low"
    sources: list[str] = Field(default_factory=list)


class NurseCount(BaseModel):
    count: int = 0
    data_source_type: DataSourceType = "simulated"
    source: str = ""


class DoctorRecord(BaseModel):
    name: str
    specialty: str = ""
    organization: str = ""
    npi: str = ""
    data_source_type: DataSourceType = "real"
    source: str = ""


class EquipmentRecord(BaseModel):
    name: str
    count: int = 0
    data_source_type: DataSourceType = "estimated"
    source: str = ""


class CanAccept(BaseModel):
    critical_patients: int = 0
    moderate_patients: int = 0
    minor_patients: int = 0


class AiRecommendation(BaseModel):
    can_accept: CanAccept = Field(default_factory=CanAccept)
    resource_gaps: list[str] = Field(default_factory=list)
    routing_advice: str = ""
    staffing_recommendations: list[str] = Field(default_factory=list)
    equipment_recommendations: list[str] = Field(default_factory=list)


class ScenarioAssignmentSection(BaseModel):
    critical_patients: FieldValue = Field(default_factory=FieldValue)
    non_critical_patients: FieldValue = Field(default_factory=FieldValue)
    total_patients: FieldValue = Field(default_factory=FieldValue)
    eta_minutes: FieldValue = Field(default_factory=FieldValue)
    distance_km: FieldValue = Field(default_factory=FieldValue)
    congestion: FieldValue = Field(default_factory=FieldValue)
    impact_zone: FieldValue = Field(default_factory=FieldValue)


class CapacitySection(BaseModel):
    scenario_assignment: ScenarioAssignmentSection = Field(default_factory=ScenarioAssignmentSection)
    total_beds: FieldValue = Field(default_factory=FieldValue)
    icu_beds: FieldValue = Field(default_factory=FieldValue)
    emergency_services: FieldValue = Field(default_factory=FieldValue)
    trauma_level: FieldValue = Field(default_factory=FieldValue)
    occupancy: FieldValue = Field(default_factory=FieldValue)


class NursesSection(BaseModel):
    er_nurses: NurseCount = Field(default_factory=NurseCount)
    icu_nurses: NurseCount = Field(default_factory=NurseCount)
    trauma_nurses: NurseCount = Field(default_factory=NurseCount)
    general_ward_nurses: NurseCount = Field(default_factory=NurseCount)


class HospitalIntelligenceResponse(BaseModel):
    hospital_profile: HospitalProfile
    capacity: CapacitySection
    doctors: list[DoctorRecord] = Field(default_factory=list)
    nurses: NursesSection = Field(default_factory=NursesSection)
    equipment: list[EquipmentRecord] = Field(default_factory=list)
    ai_recommendation: AiRecommendation = Field(default_factory=AiRecommendation)
    collected_at: str = ""
    agent_status: dict[str, str] = Field(default_factory=dict)
